compiler/types.nim

```diff
@@ -1632,6 +1635,23 @@ proc isTupleRecursive*(t: PType): bool =
   var cycleDetector = initIntSet()
   isTupleRecursive(t, cycleDetector)
 
+proc isRecursivePointer(typ: PType, isPointer: bool): bool =
+  if typ == nil:
+    return false
+  case typ.kind:
```

Comment:
No colon here and no indentation for the ``of`` branches. Just like it's done in the other 40_000 lines of code in the Nim compiler that you might have looked at by now.

Reply:
Done, it helps though if such styles would be enforced by the compiler. I remember there was a merge conflict because of this very style change in isTupleRecursive. It felt like the colon with indentation was the new correct style. That was the reason I even wrote it this way. Normally it is not my style.